extensions/ql-vscode/src/view/results/raw-results-table.tsx

```diff
@@ -19,158 +19,155 @@ import { onNavigation } from "./results";
 import { tryGetResolvableLocation } from "../../common/bqrs-utils";
 import { ScrollIntoViewHelper } from "./scroll-into-view-helper";
 import { sendTelemetry } from "../common/telemetry";
+import { assertNever } from "../../common/helpers-pure";
 
-export type RawTableProps = ResultTableProps & {
+export type RawTableProps = {
+  databaseUri: string;
   resultSet: RawTableResultSet;
   sortState?: RawResultsSortState;
   offset: number;
 };
 
-interface RawTableState {
-  selectedItem?: { row: number; column: number };
+interface TableItem {
+  readonly row: number;
+  readonly column: number;
 }
 
-export class RawTable extends React.Component<RawTableProps, RawTableState> {
-  private scroller = new ScrollIntoViewHelper();
+export function RawTable({
+  databaseUri,
+  resultSet,
+  sortState,
+  offset,
+}: RawTableProps) {
+  const [selectedItem, setSelectedItem] = useState<TableItem | undefined>();
 
-  constructor(props: RawTableProps) {
-    super(props);
-    this.setSelection = this.setSelection.bind(this);
-    this.handleNavigationEvent = this.handleNavigationEvent.bind(this);
-    this.state = {};
-  }
+  const scroller = useRef(new ScrollIntoViewHelper());
```

Comment:
This will recreate the `ScrollIntoViewHelper` on every render, even though it won't be used. [The React documentation](https://react.dev/reference/react/useRef#avoiding-recreating-the-ref-contents) shows an example of how to avoid this, so I think we should follow this:

```suggestion
  const scroller = useRef<ScrollIntoViewHelper | null>(null);
  if (scroller.current === null) {
    scroller.current = new ScrollIntoViewHelper();
  }
```

Reply:
Thanks for pointing this out.